rocAL/rocAL/source/decoders/image/fused_crop_decoder.cpp

```diff
@@ -84,62 +74,81 @@ Decoder::Status FusedCropTJDecoder::decode(unsigned char *input_buffer, size_t i
     // check the vector size for bounding box. If its more than zero go for random bbox crop
     // else go to random crop
     unsigned int crop_width, crop_height, x1, y1, x1_diff, crop_width_diff;
-    if(_bbox_coord.size() != 0)
-    {
+    if(_bbox_coord.size() != 0) {
         // Random bbox crop returns normalized crop cordinates
         // hence bringing it back to absolute cordinates 
         x1 = std::lround(_bbox_coord[0] * original_image_width);
         y1 = std::lround(_bbox_coord[1] * original_image_height);
         crop_width = std::lround((_bbox_coord[2]) * original_image_width);
         crop_height = std::lround((_bbox_coord[3]) * original_image_height);
     }
-    else
-    {
-        std::vector<float> crop_mul_param =  decoder_config.get_crop_param();
-        auto is_valid_crop = [](uint h, uint w, uint height, uint width)
-        {
-            return (h < height && w < width); 
+    else {
+        double aspect_ratio_range[2] = {decoder_config.get_random_aspect_ratio()[0], decoder_config.get_random_aspect_ratio()[1]};
+        double area_range[2] = {decoder_config.get_random_area()[0], decoder_config.get_random_aspect_ratio()[1]};
+        auto is_valid_crop = [](uint h, uint w, uint height, uint width) {
+            return(h > 0 && h <= height && w > 0 && w <= width);
         };
-        int num_of_attempts = 5;
-        for(int i = 0; i < num_of_attempts; i++)
-        {
-            double target_area  = crop_mul_param[0] * original_image_width * original_image_height;
-            crop_width  = static_cast<size_t>(std::sqrt(target_area * crop_mul_param[1]));
-            crop_height = static_cast<size_t>(std::sqrt(target_area * (1 / crop_mul_param[1])));  
-            if(is_valid_crop(crop_height, crop_width, original_image_height, original_image_width)) 
-            {
-                x1 = static_cast<size_t>(crop_mul_param[2] * (original_image_width  - crop_width));
-                y1 = static_cast<size_t>(crop_mul_param[3] * (original_image_height - crop_height));
-                break ;
+        float max_wh_ratio = aspect_ratio_range[1];
```

Comment:
assigning a double to a float value can loose precision. If so why is it passed to function as double instead of float.